piwik-sdk/src/main/java/org/piwik/sdk/CustomVariables.java

```diff
@@ -14,29 +14,26 @@
 import java.util.Arrays;
 import java.util.HashMap;
 
-
-public class CustomVariables extends HashMap<String, JSONArray> {
+/**
+ * You can track up to 5 custom variables for each user to your app,
+ * and up to 5 custom variables for each screen view.
+ * You may configure Piwik to track more custom variables: http://piwik.org/faq/how-to/faq_17931/
+ * <p/>
+ * Desired json output:
+ * {
+ * "1":["OS","iphone 5.0"],
+ * "2":["Piwik Mobile Version","1.6.2"],
+ * "3":["Locale","en::en"],
+ * "4":["Num Accounts","2"],
+ * "5":["Level","over9k"]
+ * }
+ */
+public class CustomVariables {
+    private final Object mConcurrentLock = new Object();
+    private final HashMap<String, JSONArray> mVars = new HashMap<>();
```

Comment:
I think we can go with 

```
private final Map<String, JSONArray> mVars = new ConcurrentHashMap<>();
```
and then skip the manual syncronization.
According to it's documentation the map iterator is safe during multithreading as long as each thread uses it's own iterator.